### PR TITLE
Use gvenzl/oracle-xe service

### LINKED ISSUE #5005: Use gvenzl/oracle-xe in place of wnameless/oracle-xe-11g-r2 for Oracle testing
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

[gvenzl/oracle-xe](https://hub.docker.com/r/gvenzl/oracle-xe) provides support for Oracle 11, 18 and 21, and has instructions to implement in GH Actions. Oracle 12+ would remove the limit of 30 chars for identifier length.

### PR DESCRIPTION
**Pull request by mondrake**
Fixes #5005 .

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -73,12 +73,22 @@ jobs:
         php-version:
           - "7.4"
           - "8.1"
+        oracle-version:
+          - "11"
+          - "latest"
```
Comment (morozov) [inline .github/workflows/continuous-integration.yml]: Let's use a fixed latest version as of now, e.g. 21. We will add the support for new versions explicitly.

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -73,12 +73,22 @@ jobs:
...
+        image: gvenzl/oracle-xe:${{ matrix.oracle-version }}
+        env:
+          ORACLE_PASSWORD: oracle
         ports:
           - "1521:1521"
+        options: >-
+          --health-cmd healthcheck.sh
+          --health-interval 20s
+          --health-timeout 10s
+          --health-retries 10
```
Comment (morozov) [inline .github/workflows/continuous-integration.yml]: Is it necessary to override the defaults? Usually, the defaults are sufficient as it has been with the previous image.

**Review by morozov — COMMENTED** [review]
Thanks for the patch, @mondrake. I believe it should target `2.13.x`. Otherwise, we'll have to maintain the builds based on two different images.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (mondrake) [reply .github/workflows/continuous-integration.yml]: Done

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (mondrake) [reply .github/workflows/continuous-integration.yml]: Removed

**Comment by mondrake** [comment]
Rebased and fixed.

**Comment by morozov** [comment]
Looks good in general but now it has doubled the size of the test matrix for Oracle (2 drivers × 2 PHP versions × 2 Oracle versions). I believe it's enough to test only the newest Oracle with the newest PHP (e.g. no builds for PHP 8.1 and Oracle 11).

Later, we can consider deprecating the support for Oracle 11 in DBAL 3.3 since it's been unsupported for two years ([ref](https://support.oracle.com/knowledge/Oracle%20Cloud/2068368_1.html)).

Once done, please squash all commits.

**Comment by derrabus** [comment]
> I believe it's enough to test only the newest Oracle with the newest PHP (e.g. no builds for PHP 8.1 and Oracle 11).

Shouldn't we keep at least one build with Orcale 11 for each driver, so we test with the lowest and highest supported versions? Even if we drop Oracle 11 on the 4.0.x branch, we probably want to replace it with Oracle 19 there, which is the current LTS.

**Comment by derrabus** [comment]
Suggestion:

```YAML
strategy:
  matrix:
    php-version:
      - "7.4"
      - "8.1"
    oracle-version:
      - "21"
  include:
    - php-version: "7.4"
      oracle-version: "11"
```

That would add two jobs in total for Oracle 11. If that is still too much, we could test Oracle 11 with the OCI8 driver only, so we're down to one job extra.

**Comment by morozov** [comment]
> I believe it's enough to test only the newest Oracle with the newest PHP (e.g. no builds for PHP 8.1 and Oracle 11).

What I meant is that we don't need to test the newest Oracle with all PHP versions. Not that we don't need to test the oldest Oracle version at all.

The proposal in https://github.com/doctrine/dbal/pull/5085#issuecomment-986305447 looks good.

**Comment by derrabus** [comment]
Ah, I misread your comment then. 🙂

**Comment by derrabus** [comment]
@mondrake The other way round: Oracle 21 with PHP 7.4 and 8.1, Oracle 11 with PHP 7.4 only: https://github.com/doctrine/dbal/pull/5085#issuecomment-986305447

**Review by derrabus — DISMISSED** [review]

**Comment by derrabus** [comment]
Thank you, @mondrake!

**Pull request merged**